### ISSUE TITLE
Update `@yamada-ui/table` accessibility

### ISSUE THREAD

**Issue by nakayan5** [yamada-ui/yamada-ui#1635]
### Description

his issue aims to address accessibility deficiencies in our table component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

WAI-ARIA Roles, States, and Properties
- [ ] If there is an element in the user interface that serves as a label for the table, [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) is set on the table element with a value that refers to the labelling element. Otherwise, a label is specified for the table element using [aria-label](https://w3c.github.io/aria/#aria-label).
- [ ] If the table has a caption or description, [aria-describedby](https://w3c.github.io/aria/#aria-describedby) is set on the table element with a value referring to the element containing the description.
- [x] If the table contains sortable columns or rows, [aria-sort](https://w3c.github.io/aria/#aria-sort) is set to an appropriate value on the header cell element for the sorted column or row as described in the [Grid and Table Properties Practice](https://www.w3.org/WAI/ARIA/apg/practices/grid-and-table-properties/#gridAndTableProperties_sort).
- [x] If there are conditions where some rows or columns are hidden or not present in the DOM, e.g., there are widgets on the page for hiding rows or columns, the following properties are applied as described in the [Grid and Table Properties Practice](https://www.w3.org/WAI/ARIA/apg/practices/grid-and-table-properties/).
  - [x] [aria-colcount](https://w3c.github.io/aria/#aria-colcount) or [aria-rowcount](https://w3c.github.io/aria/#aria-rowcount) is set to the total number of columns or rows, respectively.
  - [x] [aria-colindex](https://w3c.github.io/aria/#aria-colindex) or [aria-rowindex](https://w3c.github.io/aria/#aria-rowindex) is set to the position of a cell within a row or column, respectively.
- [ ] If the table includes cells that span multiple rows or multiple columns, then [aria-rowspan](https://w3c.github.io/aria/#aria-rowspan) or [aria-colspan](https://w3c.github.io/aria/#aria-colspan) is applied as described in the [Grid and Table Properties Practice](https://www.w3.org/WAI/ARIA/apg/practices/grid-and-table-properties/#gridAndTableProperties_spans).

###  Problem Statement/Justification
none

###  Proposed Solution or API
No response

###  Alternatives
none 

### Are you willing to participate in implementing this feature and create a pull request include the implementation?
None

###  Additional Information
No response

**Comment by cidkumagai** [comment]
I'll try it.

**Comment by setodeve** [comment]
@nakayan5 
Was the role unnecessary?

**Comment by nakayan5** [comment]
@setodeve 
I think it is unnecessary because of the implicit roll.
